include/slam_toolbox/merge_maps_kinetic.hpp

```diff
@@ -64,21 +58,46 @@ class MergeMapTool
   ~MergeMapTool();
 
 private:
+  typedef std::unique_ptr<karto::Pose2> Pose2_ptr;
+  typedef std::unique_ptr<karto::Vector2<kt_double>> Vector2_double_ptr;
   // setup
   void SetConfigs();
 
   // callbacks
   bool MergeMapCallback(slam_toolbox::MergeMaps::Request  &req, slam_toolbox::MergeMaps::Response &resp);
   bool AddSubmapCallback(slam_toolbox::AddSubmap::Request  &req, slam_toolbox::AddSubmap::Response &resp);
   void ProcessInteractiveFeedback(const visualization_msgs::InteractiveMarkerFeedbackConstPtr& feedback);
-  void KartoToROSOccupancyGrid(const karto::LocalizedRangeScanVector& scans);
+  void KartoToROSOccupancyGrid(const karto::LocalizedRangeScanVector& scans, nav_msgs::GetMap::Response& map);
+  void CreateNewMapper();
+  void LoadDatasetFromFile(const std::string& filename);
+
+  Pose2_ptr ApplyCorrection(const karto::Pose2& pose, const tf::Transform& submap_correction)
```

Comment:
Why are pointers needed here? Also, do you properly delete them? 
If you're going to define in the header, they should be `inline`.